### ISSUE TITLE
Using ioredis, and or sentinel/cluster modes

### ISSUE THREAD

**Issue by AVVS** [hmalphettes/redisdown#16]
Hi, I was wondering if it's possible? Judging by the code redis connection instance is created inside the module

**Comment by hmalphettes** [comment]
Hi! Thanks for raising this.
In fact it is possible to reuse a redis client that would be created else where.
```
var db = levelup('mydb', { db: redisdown, redis: redisClient });
```

Does that support the use-case?
If so I'll add a paragraph to the readme.

**Comment by AVVS** [comment]
This should cover the case for sentinel-based clients, since api is mostly compatible.

However, for cluster there are some extra restrictions so that operations that are performed on multiple keys must be performed on a single node (hashing must be used). I was wondering if there was any dependency in that matter and "restricted" operations were used?

**Comment by hmalphettes** [comment]
Again thanks for raising this and sorry I missed your quick answer.

Quick answer: place the hash inside the `location`: https://github.com/hmalphettes/redisdown#redisdownlocation

For example with the URL notation
 ```
var dbpath = '{thehash}path';
var url = "redis://user:pass@hostname:port/dbpath";
var db = levelup('dbpath', { db: redisdown, url: url });
var db = levelup(url, { db: redisdown });
```

For a given level-db database we need to keep all keys and values on the same slot of a redis cluster.
The current implementation relies on a hash to store the values and a sorted set for the keys.
It uses lua scripts to minimize each level-db calls to a single redis calls: so a single redis-call will access both key and values.

Does this support what you need?
Cheers!

**Comment by hmalphettes** [comment]
@AVVS Added a tweak to make sure the curly brackets would not be mangled in the `location` of .
Updated the tests that stopped passing with the latest `abstract-leveldown` and published a new version on npm: `0.1.7`.

Let us know how it goes!
